css/components/modal.css

```diff
@@ -0,0 +1,23 @@
+.modal-container{
+  position: absolute;
+  width:100%;
+  height:100%;
+  z-index: 1;
+  display: none;
+  background-color: var(--grey);
+  opacity: 0.7;
+  justify-content: center;
+  align-items: center;
+}
+
+.modal{
+  position: absolute;
+  z-index: 2;
+  background-color: var(--sec-col);
```

Comment:
For modal, using "position:fixed" would be the better choice instead of absolute.

The remaining code looks good to merge.